### PR TITLE
dm: add description about strict-optimistic-shard-mode

### PR DESCRIPTION
**Pull request by GMHDBJD**
### First-time contributors' checklist 

- [ ] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v7.2 (TiDB 7.2 versions)
- [ ] v7.1 (TiDB 7.1 versions)
- [ ] v7.0 (TiDB 7.0 versions)
- [ ] v6.6 (TiDB 6.6 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?
https://github.com/pingcap/tiflow/pull/9113

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: dm/feature-shard-merge-optimistic.md [hunk 1]
```diff
@@ -20,7 +20,7 @@ DM 支持在线上执行分库分表的 DDL 语句（通称 Sharding DDL），
 
 ## 乐观协调模式的配置
 
-在任务的配置文件中指定 `shard-mode` 为 `optimistic` 则使用“乐观协调”模式，示例配置文件可以参考 [DM 任务完整配置文件介绍](/dm/task-configuration-file-full.md)。
+在任务的配置文件中指定 `shard-mode` 为 `optimistic` 则使用“乐观协调”模式，可通过开启 `strict-optimistic-shard-mode` 限制“乐观协调”模式的行为，示例配置文件可以参考 [DM 任务完整配置文件介绍](/dm/task-configuration-file-full.md)。
```
Comment (ran-huang) [inline dm/feature-shard-merge-optimistic.md:23]: 这里是否要说明下有什么样的限制？

**Comment by lance6716** [comment]
lgtm

File: dm/task-configuration-file-full.md [hunk 2]
```diff
@@ -24,6 +24,7 @@ name: test                      # 任务名称，需要全局唯一
 task-mode: all                  # 任务模式，可设为 "full" - "只进行全量数据迁移"、"incremental" - "Binlog 实时同步"、"all" - "全量 + Binlog 实时同步"
 shard-mode: "pessimistic"       # 任务协调模式，可选的模式有 ""、"pessimistic、"optimistic"。默认值为 "" 即无需协调。如果是分库分表合并任务，请设置为悲观协调模式 "pessimistic"。
                                 # 在 v2.0.6 版本后乐观模式逐渐成熟，深入了解乐观协调模式的原理和使用限制后，也可以设置为乐观协调模式 "optimistic"
+strict-optimistic-shard-mode: false # 仅在乐观协调模式下生效,详见 https://docs.pingcap.com/zh/tidb/stable/feature-shard-merge-optimistic
```
Comment (ran-huang) [inline dm/task-configuration-file-full.md:27]:
```suggestion
strict-optimistic-shard-mode: false # 仅在乐观协调模式下生效，详见 https://docs.pingcap.com/zh/tidb/stable/feature-shard-merge-optimistic
```

File: dm/feature-shard-merge-optimistic.md [hunk 3]
```diff
@@ -41,7 +41,7 @@ DM 支持在线上执行分库分表的 DDL 语句（通称 Sharding DDL），
 - 增加没有默认值且非空的列：`ALTER TABLE table_name ADD COLUMN column_1 NOT NULL;`。
 - 重命名索引：`ALTER TABLE table_name RENAME INDEX index_1 TO index_2;`。
 
-各分表在执行以上 DDL 时，若顺序不同将导致同步中断，例如下述场景：
+各分表在执行以上 DDL 时，如果指定 `strict-optimistic-shard-mode: true`，会直接中断任务并报错。否则，若分表 DDL 顺序不同将导致同步中断，例如下述场景：
```
Comment (ran-huang) [inline dm/feature-shard-merge-optimistic.md:44]:
```suggestion
各分表在执行以上 DDL 时，如果指定 `strict-optimistic-shard-mode: true`，会直接中断任务并报错。如果指定 `strict-optimistic-shard-mode: false` 或未指定，若分表 DDL 顺序不同，将导致同步中断，例如下述场景：
```

File: dm/task-configuration-file-full.md [hunk 4]
```diff
@@ -24,6 +24,7 @@ name: test                      # 任务名称，需要全局唯一
 task-mode: all                  # 任务模式，可设为 "full" - "只进行全量数据迁移"、"incremental" - "Binlog 实时同步"、"all" - "全量 + Binlog 实时同步"
 shard-mode: "pessimistic"       # 任务协调模式，可选的模式有 ""、"pessimistic、"optimistic"。默认值为 "" 即无需协调。如果是分库分表合并任务，请设置为悲观协调模式 "pessimistic"。
                                 # 在 v2.0.6 版本后乐观模式逐渐成熟，深入了解乐观协调模式的原理和使用限制后，也可以设置为乐观协调模式 "optimistic"
+strict-optimistic-shard-mode: false # 仅在乐观协调模式下生效，详见 https://docs.pingcap.com/zh/tidb/stable/feature-shard-merge-optimistic
```
Comment (ran-huang) [inline dm/task-configuration-file-full.md:27]:
```suggestion
strict-optimistic-shard-mode: false # 仅在乐观协调模式下生效，限制乐观协调模式的行为，默认值为 false。详见 https://docs.pingcap.com/zh/tidb/stable/feature-shard-merge-optimistic
```

File: dm/feature-shard-merge-optimistic.md (hunk 3 above)
Comment (ran-huang) [reply dm/feature-shard-merge-optimistic.md:44]: @GMHDBJD 请看下这样修改正确吗

**Review by hfxsd — APPROVED** [review]

**Review by Frank945946 — APPROVED** [review]

**Pull request merged**